docs/source/tutorial/sampler.rst

```diff
@@ -0,0 +1,123 @@
+.. _sampler:
+
+User-Defined Sampler
+=====================
+
+Optuna allows users to create user-defined samplers.
+
+A sampler has the responsibility to determine the parameter values to be evaluated in a trial.
+When a `suggest` API (e.g., :func:`~optuna.trial.Trial.suggest_uniform`) is called inside an objective function, the corresponding distribution object (e.g., :class:`~optuna.distributions.UniformDistribution`) is created internally. A sampler samples a value from the distribution. The sampled value is returned to the caller of the `suggest` API and evaluated in the objective function.
+
+Optuna provides built-in samplers (e.g., :class:`~optuna.samplers.TPESampler`, :class:`~optuna.samplers.RandomSampler`) that work well for a wide range of cases.
+However, if you are only interested in optimizing hyperparameters in a specific domain, optimization performance may be improved if you use a sampling algorithm specialized to the domain.
```

Comment:
This sentence contains two if-clauses. Could you simplify it to improve the readability?
For example, I think we can remove the first if-clause as follows:
> However, optimization performance may be improved if you use a sampling algorithm specialized for your problem.